jobbergate-cli/jobbergate_cli/subapps/job_submissions/tools.py

```diff
@@ -98,82 +55,179 @@ def create_job_submission(
+                support=True,
+            )
+
+        job_script_files = download_job_script_files(self.job_script_id, self.jg_ctx, self.execution_directory)
 
         entrypoint_file = [f for f in job_script_files if f.file_type == FileType.ENTRYPOINT]
 
         Abort.require_condition(
             len(entrypoint_file) == 1,
-            "There should be exactly one entrypoint file in the parent job script",
+            f"There should be exactly one entrypoint file in the parent job script, got {len(entrypoint_file)}",
             raise_kwargs=dict(subject="Job Script Error"),
         )
+        job_script_path = self.execution_directory / entrypoint_file[0].filename
+
+        self.inject_sbatch_params(job_script_path)
+
+        try:
+            sbatch_handler = SubmissionHandler(
+                sbatch_path=settings.SBATCH_PATH,
+                submission_directory=self.execution_directory,
+            )
+            slurm_id = sbatch_handler.submit_job(job_script_path)
+        except Exception as e:
+            raise Abort(
+                "Failed to submit job to Slurm",
+                raise_kwargs=dict(
+                    subject="Slurm Submission Error",
+                    support=True,
+                    log_message=f"On-site submission failed: {e:s}",
+                ),
+            ) from e
+        self.slurm_job_id = slurm_id
+
+    def inject_sbatch_params(self, job_script_path: Path):
+        """Inject sbatch parameters into the job script."""
+        if not self.sbatch_arguments:
+            return
+        job_script_content = job_script_path.read_text()
+        job_script_content = inject_sbatch_params(
+            job_script_content, self.sbatch_arguments, "Injected at submission time by Jobbergate CLI"
+        )
+        job_script_path.write_text(job_script_content)
+
+    def get_request_data(self) -> JobSubmissionCreateRequestData:
+        """Return the data to be used in the POST request to the API."""
+        data = super().get_request_data()
+        data.slurm_job_id = enforce_defined(
+            getattr(self, "slurm_job_id", None),
+            "Slurm job id not found, on-site submission failed",
+            raise_exc_class=Abort,
+            raise_kwargs=dict(
+                subject="Slurm Submission Error",
+                support=True,
+            ),
+        )
+        return data
 
-        filename = entrypoint_file[0].filename
-        slurm_id = sbatch_run(filename, name)
-        job_submission_data.slurm_job_id = slurm_id
 
-    result = cast(
-        JobSubmissionResponse,
-        make_request(
-            jg_ctx.client,
-            "/jobbergate/job-submissions",
-            "POST",
-            expected_status=201,
-            abort_message="Couldn't create job submission",
-            support=True,
-            request_model=job_submission_data,
-            response_model_cls=JobSubmissionResponse,
-        ),
+def job_submissions_factory(
```

Comment:
My initial thought was to go with `*args, **kwargs` and handle them to the objects on the factory with not that much duplication. I reconsidered it to enable static type check and make it less error prone. I wonder it we could use a decorator to copy the signature from `JobSubmissionABC.__init__`, but at the same time, I have no plans to overcomplicate this.